### PR TITLE
feat: index tx by account

### PR DESCRIPTION
**Pull request by 0xdef1cafe**
## Description

wip - index txs by account for upcoming account pages

## Notice

Before submitting a pull request, please make sure you have answered the following:

- [x] Have you followed the guidelines in our [Contributing]('https://github.com/shapeshift/web/CONTRIBUTING.md) guide?
- [x] Have you checked to ensure there aren't other open [Pull Requests](https://github.com/shapeshift/web/pulls) for the same update/change?
- [ ] Do all new and existing tests pass? Does the linter pass?

## Pull Request Type

- [ ] :bug: Bug fix (Non-breaking Change: Fixes an issue)
- [ ] :hammer_and_wrench: Chore (Non-breaking Change: Doc updates, pkg upgrades, typos, etc..)
- [x] :nail_care: New Feature (Breaking/Non-breaking Change)

## Issue (if applicable)

#626 

## Testing

this won't require any testing as we're not acting on this data yet

## Screenshots (if applicable)

n/a

### CODE REVIEW

**Comment by 0xdef1cafe** [comment]
Screen Shot 2022-01-07 at 6 08 29 PM

@DaoDev44 if you want to tap away at this, i've got the tx by account in the store, tests just need fixing - see the inline comments against particular lines

File: src/state/slices/txHistorySlice/txHistorySlice.test.ts [hunk 1]
```diff
@@ -36,33 +45,61 @@ describe('txHistorySlice', () => {
...
+      )
       expect(Object.values(store.getState().txHistory.ids).length).toBe(3)
 
+      const segwitAccountSpecifier = `${BtcSend.caip2}:ypub`
+      // const segwitNativeAccountSpecifier = `${BtcSend.caip2}:zpub`
+
       // same btc transaction, different account type (send)
-      const BtcSendSegwit = { ...BtcSend, accountType: UtxoAccountType.SegwitNative }
-      store.dispatch(txHistory.actions.onMessage({ message: BtcSendSegwit }))
+      const BtcSendSegwit = { ...BtcSend, accountType: UtxoAccountType.SegwitP2sh }
```
Comment (0xdef1cafe) [inline src/state/slices/txHistorySlice/txHistorySlice.test.ts:96]: @DaoDev44 i'm pretty sure this test was wrong when i last updated it - the `txHistory.ids` length should remain the same, but we need to check that the tx id gets put against two different account Ids.

we should more generally add another test that tests tx ids against account ids to be sure it's behaving as expected.

File: src/context/TransactionsProvider/TransactionsProvider.tsx [hunk 2]
```diff
@@ -42,7 +46,19 @@ export const TransactionsProvider = ({ children }: TransactionsProviderProps): J
             await adapter.subscribeTxs(
               { wallet, accountType, ...accountParams },
               msg => {
-                dispatch(txHistory.actions.onMessage({ message: { ...msg, accountType } }))
+                const accountSpecifierObj = accountSpecifiers.reduce((acc, cur) => {
+                  if (acc) return acc
+                  const [k, v] = Object.entries(cur)[0]
+                  if (k === caip2) return v.toLowerCase()
+                  return acc
+                }, '')
```
Comment (cjthompson) [inline src/context/TransactionsProvider/TransactionsProvider.tsx:54]: this is kinda weird... can you add a comment explaining what it's doing?

**Review by cjthompson — APPROVED** [review]

File: src/context/TransactionsProvider/TransactionsProvider.tsx (hunk 2 above)
Comment (0xdef1cafe) [reply src/context/TransactionsProvider/TransactionsProvider.tsx:54]: this is a good point, it's going to work for eth but not btc correctly. 

we actually need a mapping here between accountType and x/y/z pub here; legacy is xpub, segwit is ypub, segwit native is zpub, so it should search for those accordingly.

**Review by theoboldfrazier — APPROVED** [review]

**Pull request merged**